src/aes_hash/mod.rs

```diff
@@ -160,6 +169,15 @@ impl Hasher for AHasher {
             self.hash_in(value.convert());
         } else {
             if data.len() > 32 {
+                #[cfg(all(
+                    any(target_arch = "x86", target_arch = "x86_64"),
+                    feature = "vaes",
+                    target_feature = "avx512vaes",
+                    not(miri)
+                ))]
+                if data.len() > 128 {
```

Comment:
Rather than adding another 'if' I think a cleaner way to handle this would be to add a function to factor out a method in `operations`. Like there could be `aesenc_x4` which then uses cfg to provide one implementation or the other depending on if the cpu instruction is available.